### PR TITLE
Loki: Add descriptions to query builder operations

### LINKED ISSUE #63922: Loki: Operators in query builder should have descriptions/explanations
**What happened**:
When using Loki data source, and I'm unfamiliar with LogQL so I am finding the new visual query editor very helpful, but I have no idea what these operators mean. It would be helpful to have descriptions/explanations for them.
image

Example of how do we tackle this in alerting:
image

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
adds descriptions to all operators in loki query builder
`= Equals` `!= Does not equal` `...`

there is also a visual bug fix i found whilst working on this feature.

**Why do we need this feature?**
new users to loki and logql may not know what the operators mean, for example:
`=~` or `|=`

**Who is this feature for?**
...

**Which issue(s) does this PR fix?**:
Fixes #63922

**Special notes for your reviewer**:
```
to test:
- paste the following query into the loki ds
- go back to the builder mode
- check all operators have appropriate descriptions

{compose_project=“tns-custom”} |= ip(`123`) | abc = `123` | abc = ip(`123`)
```

**feature (before & after)**
<div>
Screenshot 2023-03-02 at 15 48 50

Screenshot 2023-03-02 at 15 48 14
</div>

**bug fix (before & after)**
<div>
Screenshot 2023-03-02 at 15 50 09

Screenshot 2023-03-02 at 15 50 40
</div>

### CODE REVIEW

**Review by ivanahuckova — COMMENTED** [review]
It seems like we are repeating same things over and over. I am wondering if we should not create something like "library" for operators with its label, value and description and then always just use the one we need in each operation. That way we also ensure that the description is always correct and same. 

So something like (not sure what should be the key - if description or operator), but just to give you idea:

```
const lokiOperators = {
  equals: { label: '=', value: '=', description: 'Equals' },
};
```

**Comment by matyax** [comment]
@gwdawson Maybe you can consider something like the `explainOperator` function we use for the autocomplete, so we can reuse already existing descriptions. https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/operations.ts#L475

File: public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx [hunk 1]
```diff
@@ -98,7 +99,7 @@ export function LabelFilterItem({
             aria-label={selectors.components.QueryBuilder.matchOperatorSelect}
             value={toOption(item.op ?? defaultOp)}
             options={operators}
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx:101]:
```suggestion
            options={[
  lokiOperators.equals,
  lokiOperators.doesNotEqual,
  lokiOperators.matchesRegex,
  lokiOperators.doesNotMatchRegex,
]}
```

File: public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx [hunk 2]
```diff
@@ -160,8 +161,8 @@ export function LabelFilterItem({
...
 
 const operators = [
-  { label: '=~', value: '=~', isMultiValue: true },
-  { label: '=', value: '=', isMultiValue: false },
-  { label: '!=', value: '!=', isMultiValue: false },
-  { label: '!~', value: '!~', isMultiValue: true },
+  lokiOperators.equals,
+  lokiOperators.doesNotEqual,
+  lokiOperators.matchesRegex,
+  lokiOperators.doesNotMatchRegex,
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx:167]: I'd remove this and just include operators as options.

File: public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx [hunk 3]
```diff
@@ -98,7 +99,7 @@ export function LabelFilterItem({
             aria-label={selectors.components.QueryBuilder.matchOperatorSelect}
             value={toOption(item.op ?? defaultOp)}
             options={operators}
-            width="auto"
+            width={14}
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx:102]: Here specifically - I personally like `auto` option more than `14`, cause it reads/looks like query. But that's just me. It is worth checking with logs and metrics team (as this is going to change UI for prometheus as well) what they think. 
`auto`: 
image
image

`14`:
image
image

**Review by ivanahuckova — COMMENTED** [review]
Great job! I'd check with metrics team what they think about this change as the Label filter UI is going to be changed for Prometheus as well.

File: public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx (hunk 1 above)
Comment (gwdawson) [reply public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx:101]: this is also used on `line 45` so i think it makes sense to keep it here and have them in sync.

File: public/app/plugins/datasource/prometheus/querybuilder/shared/OperationEditor.tsx [hunk 4]
```diff
@@ -245,6 +245,9 @@ function callParamChangedThenOnChange(
 
 const getStyles = (theme: GrafanaTheme2) => {
   return {
+    cardWrapper: css({
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/OperationEditor.tsx:248]: last question, why are we adding `cardWrapper` and `height: '100%'`. I tested UI without these changes and I don't see any difference. So I am probably missing something, just curious.

**Review by ivanahuckova — COMMENTED** [review]
1 last question. Otherwise LGTM 🔥 !

File: public/app/plugins/datasource/prometheus/querybuilder/shared/OperationEditor.tsx (hunk 4 above)
Comment (gwdawson) [reply public/app/plugins/datasource/prometheus/querybuilder/shared/OperationEditor.tsx:248]: this actually fixes a bug that i realised i added in a previous PR.
ideally i should have made a separate pr for this but i found it whilst working on this so i decided to fix it here.

**with**
Screenshot 2023-03-14 at 09 58 33


**without**
Screenshot 2023-03-14 at 09 58 57

**Review by ivanahuckova — APPROVED** [review]
LGTM

**Pull request merged**